### PR TITLE
TW-264: fix swap display output calculation

### PR DESCRIPTION
**Pull request by letier3110**

### CODE REVIEW

File: src/screens/swap/swap-form/swap-form.tsx [hunk 1]
```diff
@@ -167,18 +170,43 @@ export const SwapForm: FC<SwapFormProps> = ({ inputToken }) => {
...
+    }
+
+    return undefined;
+  }, [bestTradeWithSlippageTolerance, outputAssets.asset.decimals]);
+
+  useEffect(() => {
+    if (isDefined(bestTrade)) {
+      const bestTradeOutput = getTradeOutputAmount(bestTrade);
+
+      const outputTzAmount = bestTradeOutput ? mutezToTz(bestTradeOutput, outputAssets.asset.decimals) : undefined;
```
Comment (LendiHop) [inline src/screens/swap/swap-form/swap-form.tsx:187]: undefined => new BigNumber(0)

File: src/screens/swap/swap-form/swap-form.tsx [hunk 2]
```diff
@@ -167,18 +170,43 @@ export const SwapForm: FC<SwapFormProps> = ({ inputToken }) => {
...
+
+      return mutezToTz(lastTradeOperation.bTokenAmount, outputAssets.asset.decimals);
+    }
+
+    return undefined;
+  }, [bestTradeWithSlippageTolerance, outputAssets.asset.decimals]);
+
+  useEffect(() => {
+    if (isDefined(bestTrade)) {
+      const bestTradeOutput = getTradeOutputAmount(bestTrade);
```
Comment (LendiHop) [inline src/screens/swap/swap-form/swap-form.tsx:185]: move this all logic to getTradeOutputAmount()

File: src/screens/swap/swap-form/swap-form.tsx [hunk 3]
```diff
@@ -167,18 +170,43 @@ export const SwapForm: FC<SwapFormProps> = ({ inputToken }) => {
...
+
+      const outputTzAmount = bestTradeOutput ? mutezToTz(bestTradeOutput, outputAssets.asset.decimals) : undefined;
+
+      const feeAmount = minimumReceivedAmount?.minus(minimumReceivedAmount?.multipliedBy(ROUTING_FEE_RATIO));
+      const finalAmount = outputTzAmount?.minus(feeAmount ?? new BigNumber(0));
+
+      setFieldValue(
+        'outputAssets.amount',
+        tzToMutez(finalAmount ?? new BigNumber(0), outputAssets.asset.decimals),
```
Comment (LendiHop) [inline src/screens/swap/swap-form/swap-form.tsx:194]: 2. rm fallback

File: src/screens/swap/swap-form/swap-form.tsx (hunk 2 above)
Comment (letier3110) [reply src/screens/swap/swap-form/swap-form.tsx:185]: can't modify this fn due to different use-cases in exchange rate calculation and output calculation

**Review by LendiHop — DISMISSED** [review]

**Review by NovakVlad2424 — DISMISSED** [review]

File: src/screens/swap/swap-form/swap-form.tsx [hunk 4]
```diff
@@ -167,18 +170,41 @@ export const SwapForm: FC<SwapFormProps> = ({ inputToken }) => {
...
+        ? mutezToTz(bestTradeOutput, outputAssets.asset.decimals)
+        : new BigNumber(0);
+
+      const feeAmount = minimumReceivedAmount.minus(minimumReceivedAmount.multipliedBy(ROUTING_FEE_RATIO));
+      const finalAmount = outputTzAmount.minus(feeAmount);
+
+      setFieldValue('outputAssets.amount', tzToMutez(finalAmount, outputAssets.asset.decimals), false);
+    }
+  }, [bestTrade, minimumReceivedAmount, outputAssets.asset.decimals, outputAssetSlug, setFieldValue]);
+
```
Comment (unixvb) [inline src/screens/swap/swap-form/swap-form.tsx:197]: the same comments that you  have on extension 

https://github.com/madfish-solutions/templewallet-extension/pull/755

**Review by LendiHop — APPROVED** [review]

**Review by LendiHop — DISMISSED** [review]

File: src/screens/swap/swap-form/swap-form.tsx [hunk 5]
```diff
@@ -250,7 +277,7 @@ export const SwapForm: FC<SwapFormProps> = ({ inputToken }) => {
               inputAssets={inputAssets}
               outputAssets={outputAssets}
               bestTrade={bestTrade}
-              bestTradeWithSlippageTolerance={bestTradeWithSlippageTolerance}
+              minimumReceivedAmount={minimumReceivedAmount}
```
Comment (unixvb) [inline src/screens/swap/swap-form/swap-form.tsx:280]: fee is not included in this "minimumReceivedAmount"

**Pull request closed without merging**